location/src/iosMain/kotlin/location/LocationMonitor.kt

```diff
@@ -0,0 +1,71 @@
+/*
+ Copyright 2021 Splendo Consulting B.V. The Netherlands
+ 
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+ 
+      http://www.apache.org/licenses/LICENSE-2.0
+ 
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+  
+ */
+
+package com.splendo.kaluga.location
+
+import com.splendo.kaluga.base.ServiceMonitor
+import com.splendo.kaluga.logging.debug
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import platform.CoreLocation.CLLocationManager
+import platform.CoreLocation.CLLocationManagerDelegateProtocol
+import platform.darwin.NSObject
+
+actual class LocationMonitor(private val locationManager: CLLocationManager) : ServiceMonitor {
+
+    private companion object {
+        private const val TAG = "LocationMonitor"
+    }
+
+    actual class Builder {
+        actual fun create(): LocationMonitor = LocationMonitor(
+            locationManager = CLLocationManager()
+        )
+    }
+
+    internal class LocationManagerDelegate(
+        private val updateState: () -> Unit
+    ) : NSObject(), CLLocationManagerDelegateProtocol {
+        override fun locationManagerDidChangeAuthorization(manager: CLLocationManager) {
+            debug(TAG) { "locationManagerDidChangeAuthorization" }
+            updateState()
+        }
+    }
+
+
+    private val isLocationEnabled: Boolean
+        get() = locationManager.locationServicesEnabled()
+
+    private val _isEnabled = MutableStateFlow(false)
```

Comment:
should this not be `isLocationEnabled`?

Reply:
you could make a abstract base class with `abstract protected val isLocationEnabled: Boolean` and the stateflows already there to avoid this mistake.

Reply:
in fact there could probably an abstract "monitor" base class and interface, I assume most of this code is copy paste from the other monitor?

Reply:
Right, forgot to change the initial value for the mutable state.
>should this not be isLocationEnabled?

I created `ServiceMonitor` interface, but I didn't use it in BluetoothMonitor yet. 
> in fact there could probably be an abstract "monitor" base class and interface

Reply:
11244f6